### ISSUE TITLE
Arithmetic operation resulted in an overflow

### ISSUE THREAD

**Issue by kentcb** [Microsoft/service-fabric-client-dotnet#53]
## Describe the bug
With `2.0.0-preview1`, when attempting to upload an application package using the image store, I get:
```
System.OverflowException: Arithmetic operation resulted in an overflow.
   at Microsoft.ServiceFabric.Client.Http.ImageStoreClient.<UploadChunkAsync>d__17.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.ServiceFabric.Common.Utilities.ConcurrentOperationsRunner`1.<RunOperationsSerially>d__6.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.ServiceFabric.Common.Utilities.ConcurrentOperationsRunner`1.<RunAll>d__5.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.ServiceFabric.Client.Http.ImageStoreClient.<UploadAllChunksAsync>d__16.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task task)
   at Microsoft.ServiceFabric.Client.Http.ImageStoreClient.<UploadApplicationPackageAsync>d__9.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.GetResult()
   at UserQuery.<Main>d__0.MoveNext() in C:\Users\Kent\AppData\Local\Temp\LINQPad5\_djbahaxc\query_mwrirh.cs:line 646
```

## To Reproduce
Here's a LINQPad script:
```C#
async Task Main()
{
    var automationCertificate = new X509Certificate2(@"C:\Users\Kent\Repository\<<redacted>>\Azure.pfx");
    var clusterManagementEndpoint = "https://<<redacted>>cluster.northcentralus.cloudapp.azure.com:19080";
    var clusterManagementUri = new Uri(clusterManagementEndpoint);

    var fabricClient = await new ServiceFabricClientBuilder()
        .UseEndpoints(clusterManagementUri)
        .UseX509Security(
            _ =>
            {
                var remoteSecuritySettings = new RemoteX509SecuritySettings(new List<string> { "<<redacted>>" });
                var x509SecuritySettings = new X509SecuritySettings(automationCertificate, remoteSecuritySettings);
                return System.Threading.Tasks.Task.FromResult<SecuritySettings>(x509SecuritySettings);
            })
        .BuildAsync();

    "Waiting for cluster management endpoint to become available.".Dump();

    while (true)
    {
        try
        {
            var clusterHealth = await fabricClient.Cluster.GetClusterHealthAsync();

            if (clusterHealth.AggregatedHealthState == HealthState.Ok)
            {
                "Connected to cluster management endpoint.".Dump();
                break;
            }

            $"Cluster health is not OK. It is '{clusterHealth.AggregatedHealthState}' - waiting.".Dump();
        }
        catch (ServiceFabricRequestException)
        {
            "Unable to connect to Service Fabric management endpoint - waiting.".Dump();
        }

        await System.Threading.Tasks.Task.Delay(TimeSpan.FromSeconds(10));
    }
    
    var resourceNamePrefix = "<<redacted>>";
    var pkgDir = @"C:\Users\Kent\Repository\<<redacted>>\pkg\Release\";

    "Uploading package to image store.".Dump();
    try
    {
        await fabricClient
            .ImageStore
            .UploadApplicationPackageAsync(pkgDir, compressPackage: true, applicationPackagePathInImageStore: resourceNamePrefix);
    }
    catch (Exception ex)
    {
        // THIS IS TRIGGERING
        ex.ToString().Dump();
    }

    "Provisioning application type.".Dump();
    var applicationTypeDescription = new ProvisionApplicationTypeDescription(
        resourceNamePrefix,
        applicationPackageCleanupPolicy: ApplicationPackageCleanupPolicy.Automatic);
    await fabricClient
        .ApplicationTypes
        .ProvisionApplicationTypeAsync(applicationTypeDescription);

    "Creating application.".Dump();
    var applicationDescription = new ApplicationDescription(
        new ApplicationName("fabric:/" + resourceNamePrefix),
        resourceNamePrefix,
        "1.0.0");
    await fabricClient
        .Applications
        .CreateApplicationAsync(applicationDescription);

    $"Done".Dump();
}
```

I have tried both with and without compression enabled - the result is the same.

## Expected behavior
The package should upload without error, or provide better information about why it is failing.

## Additional context
I'm attempting to automate SF deployments. I've repro'd this on both Linux and Windows, though I may also have an _additional_ problem on Linux to do with SSL (will file a separate issue if it comes to it).

I'd be happy to provide the package itself privately if that helps.

**Comment by oanapl** [comment]
@kentcb , thank you for reporting, we appreciate your help here. 
Adding @iyyappam , who will investigate this on our side.

**Comment by iyyappam** [comment]
@kentcb, thanks for reporting the issue. Could you share the client side traces? If you are running local cluster on Windows, the traces will be created on C:\sfDevcluster\Logs\Traces. Please provide the timestamp of the issue.

**Comment by kentcb** [comment]
Dumb question, but do those traces only populate if deploying to a local cluster? I just deleted existing traces, reproduced the problem (against Azure proper) and there are no traces output in the folder. :/

**Comment by iyyappam** [comment]
No, you can keep the local cluster running and deploy to the remote cluster to repro the scenario. The client side traces still will be generated. The key here is local cluster should be running (just installed is alone not sufficient).

**Comment by kentcb** [comment]
Ugh, not having much luck with it. I deleted the existing traces again because I forgot to record the time I repro'd, then I repro'd again and noted the time, but it's been over six _hours_ and nothing has appeared :confused:

**Comment by iyyappam** [comment]
You could try one more thing to flush the traces. 

1. Make sure the local cluster is running (and repro happened).

2. Then go to Computer Management -> System Tools -> Performance -> Data Collector Sets -> Event Trace Trace Session -> Fabric Traces

3. Now stop the Fabric Traces (triggers flushing to disk) and wait for the traces to appear in C:\sfdevcluster\Log\Traces.

4. Fabric Traces should automatically running again in few moments, else you can manually start the session.

**Comment by amanbha** [comment]
@kentcb  Looks like some issue in calulating chunk size to upload on the client side, Could you share the sizes of files in your package which are greater than 25 * 1024 * 1024 bytes.

**Comment by amanbha** [comment]
Fixed in 2.0.0-preview11 version of nuget package
